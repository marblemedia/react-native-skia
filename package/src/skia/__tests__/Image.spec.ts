import { loadImage } from "../../renderer/__tests__/setup";
import { ColorType, AlphaType } from "../types";

import { setupSkia } from "./setup";

describe("Image", () => {
  it("readPixels", () => {
    setupSkia();
    const image = loadImage("skia/__tests__/assets/oslo.jpg");
    const pixels = image.readPixels(0, 0, {
      width: 2,
      height: 2,
      colorType: ColorType.RGBA_8888,
      alphaType: AlphaType.Unpremul,
    });
    expect(pixels).toMatchSnapshot();
  });
});
