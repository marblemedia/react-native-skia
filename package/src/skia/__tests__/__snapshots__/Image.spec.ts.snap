// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image readPixels 1`] = `
Uint8Array [
  170,
  186,
  199,
  255,
  170,
  186,
  199,
  255,
  170,
  186,
  199,
  255,
  170,
  186,
  199,
  255,
]
`;
